pyproject.toml

```diff
@@ -17,26 +17,26 @@ keywords = [
   "science",
 ]
 license = {file = "LICENSE.md"}
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 dependencies = [
   "astropy>=5.0.1",
   "cached-property>=1.5.2",
-  "h5py>=3",
+  "h5py>=3.1",
   "ipywidgets>=7.6.5",
   "lmfit>=1",
   "matplotlib>=3.3",
   "mpmath>=1.2.1",
   "numba",
-  "numpy>=1.20",
+  "numpy>=1.21",
   "packaging",
-  "pandas>=1",
+  "pandas>=1.2",
```

Comment:
I think it was the `pandas` dependency that was causing problems on the tests with the minimum versions.  For most of the dependencies that I changed, I looked for a release that was roughly two years old and set that as the minimum version, which is what NEP 29 recommends doing for NumPy and is probably a good balance between reducing maintenance burden and having a wide enough window of allowed package versions.